app/controllers/api_controller/tags.rb

```diff
@@ -28,37 +28,25 @@ def tags_unassign_resource(object, _type, id = nil, data = nil)
 
     def tags_create_resource(parent, _type, _id, data)
       entry = parent.add_entry(data)
-      raise BadRequestError, "#{entry.errors.full_messages.join(', ')}" unless entry.valid?
+      raise_failed_to_add_resource_error(:tag, entry) unless entry.valid?
       entry.tag
-    rescue => err
-      raise BadRequestError, "Could not create a new tag - #{err}"
     end
 
     def create_resource_tags(_type, _id, data)
-      if data.key?("id") || data.key?("href")
-        raise BadRequestError,
-              "Resource id or href should not be specified for creating a new tag resource"
-      end
+      raise_resource_id_or_href_specified_error(:tag) if data.key?("id") || data.key?("href")
       category_data = data.delete("category") { {} }
       category = fetch_category(category_data)
-      unless category
-        category_rep = category_data.map { |k, v| "#{k} = #{v}" }.join(', ')
-        raise BadRequestError, "Could not find category with data #{category_rep}"
-      end
-      begin
-        entry = category.add_entry(data)
-        raise BadRequestError, "#{entry.errors.full_messages.join(', ')}" unless entry.valid?
-        entry.tag
-      rescue => err
-        raise BadRequestError, "Could not create a new tag - #{err}"
-      end
+      raise_could_not_find_resource_error(:category, category_data) unless category
+      entry = category.add_entry(data)
+      raise_failed_to_add_resource_error(:tag, entry) unless entry.valid?
+      entry.tag
```

Comment:
This was somewhat deliberate as I noticed we don't do this in the generic method. I also couldn't remember why we had added that - we are rescuing a `BadRequestError` only to re-raise it with another `BadRequestError`?